### PR TITLE
Add direction & throttler mode accessors to improve Java interop

### PR DESCRIPTION
**Pull request by vleletko**
#25429
added acessors to ThrottlerTransportAdapter object, 
added compile only java tests.

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
Hi @vleletko,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

**Comment by ktoso** [comment]
OK TO TEST

**Review by ktoso — APPROVED** [review]
Looks good, thanks!

File: akka-remote/src/main/scala/akka/remote/transport/ThrottlerTransportAdapter.scala [hunk 1]
```diff
@@ -165,6 +165,31 @@ object ThrottlerTransportAdapter {
...
+
+  /**
+   * Java API: get the ThrottleMode.Blackhole instance
+   */
+  def blackholeThrottleMode() = Blackhole
+
+  /**
+   * Java API: get the ThrottleMode.Unthrottled instance
+   */
+  def unthrottledThrottleMode() = Unthrottled
```
Comment (patriknw) [inline akka-remote/src/main/scala/akka/remote/transport/ThrottlerTransportAdapter.scala]: I'd have preferred explicit return types so added them in https://github.com/akka/akka/pull/25474

**Pull request merged**